geode-core/src/main/java/org/apache/geode/internal/cache/partitioned/PersistentBucketRecoverer.java

```diff
@@ -182,12 +186,23 @@ public void run2() {
         if (!warningLogged) {
           sleepMillis = SLEEP_PERIOD / 2;
         }
+        if (regions.isEmpty()) {
+          break;
+        }
         Thread.sleep(sleepMillis);
 
         if (membershipChanged) {
           membershipChanged = false;
           for (RegionStatus region : regions) {
-            region.logWaitingForMembers();
+            try {
```

Comment:
Instead of having a destroyedRegions collection, I think it would be better to use an Iterator on regions. Then in the catch you can just call iterator.remove.

Reply:
Thanks for comments. Updated